[ Skip to content ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graphexceptions>)
[ ![logo](https://ai.pydantic.dev/img/logo-white.svg) ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/..> "PydanticAI")
PydanticAI 
pydantic_graph.exceptions 
Initializing search 
[ pydantic/pydantic-ai  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/github.com/pydantic/pydantic-ai> "Go to repository")
[ ![logo](https://ai.pydantic.dev/img/logo-white.svg) ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/..> "PydanticAI") PydanticAI 
[ pydantic/pydantic-ai  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/github.com/pydantic/pydantic-ai> "Go to repository")
  * [ Introduction  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/..>)
  * [ Installation  ](https://ai.pydantic.dev/api/pydantic_graph/install/>)
  * [ Getting Help  ](https://ai.pydantic.dev/api/pydantic_graph/help/>)
  * [ Contributing  ](https://ai.pydantic.dev/api/pydantic_graph/contributing/>)
  * [ Troubleshooting  ](https://ai.pydantic.dev/api/pydantic_graph/troubleshooting/>)
  * Documentation  Documentation 
    * [ Agents  ](https://ai.pydantic.dev/api/pydantic_graph/agents/>)
    * [ Models  ](https://ai.pydantic.dev/api/pydantic_graph/models/>)
    * [ Dependencies  ](https://ai.pydantic.dev/api/pydantic_graph/dependencies/>)
    * [ Function Tools  ](https://ai.pydantic.dev/api/pydantic_graph/tools/>)
    * [ Results  ](https://ai.pydantic.dev/api/pydantic_graph/results/>)
    * [ Messages and chat history  ](https://ai.pydantic.dev/api/pydantic_graph/message-history/>)
    * [ Testing and Evals  ](https://ai.pydantic.dev/api/pydantic_graph/testing-evals/>)
    * [ Debugging and Monitoring  ](https://ai.pydantic.dev/api/pydantic_graph/logfire/>)
    * [ Multi-agent Applications  ](https://ai.pydantic.dev/api/pydantic_graph/multi-agent-applications/>)
    * [ Graphs  ](https://ai.pydantic.dev/api/pydantic_graph/graph/>)
  * [ Examples  ](https://ai.pydantic.dev/api/pydantic_graph/examples/>)
Examples 
    * [ Pydantic Model  ](https://ai.pydantic.dev/api/pydantic_graph/examples/pydantic-model/>)
    * [ Weather agent  ](https://ai.pydantic.dev/api/pydantic_graph/examples/weather-agent/>)
    * [ Bank support  ](https://ai.pydantic.dev/api/pydantic_graph/examples/bank-support/>)
    * [ SQL Generation  ](https://ai.pydantic.dev/api/pydantic_graph/examples/sql-gen/>)
    * [ Flight booking  ](https://ai.pydantic.dev/api/pydantic_graph/examples/flight-booking/>)
    * [ RAG  ](https://ai.pydantic.dev/api/pydantic_graph/examples/rag/>)
    * [ Stream markdown  ](https://ai.pydantic.dev/api/pydantic_graph/examples/stream-markdown/>)
    * [ Stream whales  ](https://ai.pydantic.dev/api/pydantic_graph/examples/stream-whales/>)
    * [ Chat App with FastAPI  ](https://ai.pydantic.dev/api/pydantic_graph/examples/chat-app/>)
    * [ Question Graph  ](https://ai.pydantic.dev/api/pydantic_graph/examples/question-graph/>)
  * API Reference  API Reference 
    * [ pydantic_ai.agent  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/agent/>)
    * [ pydantic_ai.tools  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/tools/>)
    * [ pydantic_ai.result  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/result/>)
    * [ pydantic_ai.messages  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/messages/>)
    * [ pydantic_ai.exceptions  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/exceptions/>)
    * [ pydantic_ai.settings  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/settings/>)
    * [ pydantic_ai.usage  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/usage/>)
    * [ pydantic_ai.format_as_xml  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/format_as_xml/>)
    * [ pydantic_ai.models  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/base/>)
    * [ pydantic_ai.models.openai  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/openai/>)
    * [ pydantic_ai.models.anthropic  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/anthropic/>)
    * [ pydantic_ai.models.cohere  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/cohere/>)
    * [ pydantic_ai.models.gemini  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/gemini/>)
    * [ pydantic_ai.models.vertexai  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/vertexai/>)
    * [ pydantic_ai.models.groq  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/groq/>)
    * [ pydantic_ai.models.mistral  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/mistral/>)
    * [ pydantic_ai.models.test  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/test/>)
    * [ pydantic_ai.models.function  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/models/function/>)
    * [ pydantic_graph  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<../graph/>)
    * [ pydantic_graph.nodes  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<../nodes/>)
    * [ pydantic_graph.state  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<../state/>)
    * [ pydantic_graph.mermaid  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<../mermaid/>)
    * pydantic_graph.exceptions  [ pydantic_graph.exceptions  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<./>) Table of contents 
      * [ exceptions  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions>)
      * [ GraphSetupError  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphSetupError>)
        * [ message  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphSetupError.message>)
      * [ GraphRuntimeError  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphRuntimeError>)
        * [ message  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphRuntimeError.message>)


Table of contents 
  * [ exceptions  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions>)
  * [ GraphSetupError  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphSetupError>)
    * [ message  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphSetupError.message>)
  * [ GraphRuntimeError  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphRuntimeError>)
    * [ message  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphRuntimeError.message>)


  1. [ Introduction  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/..>)
  2. [ API Reference  ](https://ai.pydantic.dev/api/pydantic_graph/exceptions/agent/>)


Version Notice
This documentation is ahead of the last release by [1 commit](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/github.com/pydantic/pydantic-ai/compare/v0.0.21...main>). You may see documentation for features not yet supported in the latest release [v0.0.21 2025-01-30](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/github.com/pydantic/pydantic-ai/releases/tag/v0.0.21>). 
# `pydantic_graph.exceptions`
###  GraphSetupError
Bases: `TypeError[](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/docs.python.org/3/library/exceptions.html#TypeError>)`
Error caused by an incorrectly configured graph.
Source code in `pydantic_graph/pydantic_graph/exceptions.py`
```
1
2
3
4
5
6
7
8
9
```
| ```
class GraphSetupError(TypeError):
"""Error caused by an incorrectly configured graph."""
  message: str
"""Description of the mistake."""
  def __init__(self, message: str):
    self.message = message
    super().__init__(message)

```
  
---|---  
####  message `instance-attribute`
```
message: str[](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/docs.python.org/3/library/stdtypes.html#str>) = message[](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphSetupError.message> "pydantic_graph.exceptions.GraphSetupError.message")

```

Description of the mistake.
###  GraphRuntimeError
Bases: `RuntimeError[](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/docs.python.org/3/library/exceptions.html#RuntimeError>)`
Error caused by an issue during graph execution.
Source code in `pydantic_graph/pydantic_graph/exceptions.py`
```
12
13
14
15
16
17
18
19
20
```
| ```
class GraphRuntimeError(RuntimeError):
"""Error caused by an issue during graph execution."""
  message: str
"""The error message."""
  def __init__(self, message: str):
    self.message = message
    super().__init__(message)

```
  
---|---  
####  message `instance-attribute`
```
message: str[](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<https:/docs.python.org/3/library/stdtypes.html#str>) = message[](https://ai.pydantic.dev/api/pydantic_graph/exceptions/<#pydantic_graph.exceptions.GraphRuntimeError.message> "pydantic_graph.exceptions.GraphRuntimeError.message")

```

The error message.
Â© Pydantic Services Inc. 2024 to present 
